framework/harness.go

```diff
@@ -0,0 +1,135 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+)
+
+const httpListenerTimeout = time.Second * 10
+
+type TestHarness struct {
+	testServiceBaseURL         string
+	testHarnessExternalBaseURL string
+	testServiceInfo            TestServiceInfo
+	mockEndpoints              *mockEndpointsManager
+	logger                     Logger
+	lock                       sync.Mutex
+}
+
+// NewTestServiceClient creates a TestServiceClient instance, and verifies that the test service
```

Comment:
Update comment to reflect the new function name (NewTestServiceClient -> NewTestHarness).